core/iwasm/libraries/lib-wasi-threads/stress_test/mutex_common.h

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef MUTEX_COMMON_H
+#define MUTEX_COMMON_H
+
+#include <pthread.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+typedef struct {
+    pthread_mutex_t *mutex;
+    int *counter;
```

Comment:
does it have to be a pointer? Based on the usage in the code, this can be just an int value.

Reply:
well, I'm afraid to be wrong, but this variable is passed to threads as a part of *arg and updated there.

If it'd be a value, every thread would receive a copy and we wouldn't have effective multithreading, because counter here is effectively a shared resource, and as it's not global var I'm passing a pointer. 

Please correct me if I'm wrong

Reply:
don't you pass a pointer to the whole `MutexCounter` struct?

Reply:
You're extremely right, sorry 😂

Reply:
well  do but later I put it inside a var 

`    MutexCounter mutex_counter = *(MutexCounter *)(arg); `

It's confusing so I will refactor it probably